### PR TITLE
[tune] Allow perturbations of categorical variables

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

Rather than always resampling, this allows categorical variables to be perturbed to adjacent values.

### CODE REVIEW

File: python/ray/tune/pbt.py [hunk 1]
```diff
@@ -130,11 +138,12 @@ class PopulationBasedTraining(FIFOScheduler):
...
         >>>     hyperparam_mutations={
-        >>>         # Allow for scaling-based perturbations, with a uniform
-        >>>         # backing distribution for resampling.
-        >>>         "factor_1": lambda config: random.uniform(0.0, 20.0),
-        >>>         # Only allows resampling from this list as a perturbation.
-        >>>         "factor_2": [1, 2],
+        >>>         # Perturb factor1 by scaling it by 0.8 or 1.2. Resampling
+        >>>         # resets it to a value sampled from the lambda function.
+        >>>         "factor_1": lambda: random.uniform(0.0, 20.0),
+        >>>         # Perturb factor2 by changing it to aan djacent value, e.g.
```
Comment (richardliaw) [inline python/ray/tune/pbt.py]: typo

**Review by richardliaw — COMMENTED** [review]
Is this in the paper?

**Comment by ericl** [comment]
This isn't in the paper I think but is much more convenient in practice to allow parameters to vary over a discrete set of values.

File: python/ray/tune/pbt.py (hunk 1 above)
Comment (ericl) [reply python/ray/tune/pbt.py]: Fixed

File: python/ray/rllib/es/es.py [hunk 2]
```diff
@@ -300,6 +300,10 @@ def _train(self):
 
         return result
 
+    def _stop(self):
```
[the remark this answers is not in the document]
Comment (ericl) [reply python/ray/rllib/es/es.py]: Moved to another pr.

**Review by richardliaw — APPROVED** [review]
Looks fine, can merge ater last test is run

**Pull request merged**